### PR TITLE
Feat/liked videos page

### PR DESCRIPTION
**Pull request by BalajiHarinath**
I have added history page and liked video page. Please review.

Files to be reviewed:
[context/HistoryContext.js](https://github.com/BalajiHarinath/Video-library-react/blob/feat/liked-videos-page/src/context/HistoryContext.js)
[context/LikedVideoContext.js](https://github.com/BalajiHarinath/Video-library-react/blob/feat/liked-videos-page/src/context/LikedVideoContext.js)
[components/CardHistory/CardHistory.jsx](https://github.com/BalajiHarinath/Video-library-react/blob/feat/liked-videos-page/src/components/CardHistory/CardHistory.jsx)
[components/CardLikedVideo/CardLikedVideo.jsx](https://github.com/BalajiHarinath/Video-library-react/blob/feat/liked-videos-page/src/components/CardLikedVideo/CardLikedVideo.jsx)
[pages/History/History.jsx](https://github.com/BalajiHarinath/Video-library-react/blob/feat/liked-videos-page/src/pages/History/History.jsx)
[pages/History/History.css](https://github.com/BalajiHarinath/Video-library-react/blob/feat/liked-videos-page/src/pages/History/History.css)
[pages/Liked/LikedVideos.jsx](https://github.com/BalajiHarinath/Video-library-react/blob/feat/liked-videos-page/src/pages/Liked/LikedVideos.jsx)
[pages/Liked/LikedVideos.css](https://github.com/BalajiHarinath/Video-library-react/blob/feat/liked-videos-page/src/pages/Liked/LikedVideos.css)

[Live Link](https://easytv-liked-videos.netlify.app/)

### CODE REVIEW

File: src/components/CardHistory/CardHistory.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+  const { removeVideoFromHistory } = useHistory();
+
+  useEffect(() => {
+    const clickHandler = () => {
+      if (isMoreOptions) {
+        setIsMoreOptions(false);
+      }
+    };
+    document.addEventListener("click", clickHandler);
```
Comment (anik31) [inline src/components/CardHistory/CardHistory.jsx:20]: suggestion - we should avoid using document.addEventListener in react. Was it required here ?

File: src/components/CardLikedVideo/CardLikedVideo.jsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                  watch_later
+                </span>
+                <div className="btn-transparent text-sm">
+                  Add to Watch Later
+                </div>
+              </li>
+            )}
+
+            {LikedVideos.length > 0 &&
+            LikedVideos.some((item) => item._id === _id) ? (
```
Comment (anik31) [inline src/components/CardLikedVideo/CardLikedVideo.jsx:116]: appreciation - good that you're using array.some

File: src/components/SidebarNav/SidebarNavData.js [hunk 3]
```diff
@@ -40,6 +40,6 @@ export const SidebarNavData = [
     _id: uuid(),
     name: "History",
     icon: "history",
-    link: "./adscad",
+    link: "/history",
   },
```
Comment (anik31) [inline src/components/SidebarNav/SidebarNavData.js:44]: appreciation - good that you're using an array of objects to store sidebar data

**Review by anik31 — COMMENTED** [review]
Nice work 👍 Everything looks good

**Review by abhisheksahani-nova — APPROVED** [review]
Clean and readable code looks good to merge, great work

File: src/components/CardHistory/CardHistory.jsx (hunk 1 above)
Comment (BalajiHarinath) [reply src/components/CardHistory/CardHistory.jsx:20]: Yes I tried only using react but could not achieve what I wanted. This is basically to hide the more options div when clicked anywhere on the page or on the more options button again.

**Comment by BalajiHarinath** [comment]
Thanks for the review. Merging it.

**Pull request merged**